src/cursor-doc/paredit.ts

```diff
@@ -138,6 +138,49 @@ export function backwardListRange(doc: EditableDocument, start: number = doc.sel
     return [cursor.offsetStart, start];
 }
 
+
+/**
+ * Aims to find the end of the current form (list|vector|map|set|string etc) or
+ * the first newline and delete up to the newline or the form's closing delimiter.
+ * When the first newline is in the middle of another form then the end
+ * of the range is the end of the form that includes the first newline.
+ *
+ * This function's output range is needed to implement features similar to paredit's
+ * killRight or smartparens' sp-kill-hybrid-sexp.
+ *
+ * @param doc
+ * @param offset
+ * @param goPastWhitespace
+ * @returns [number, number]
+ */
+export function forwardHybridSexpRange(doc: EditableDocument, offset = Math.max(doc.selection.anchor, doc.selection.active), goPastWhitespace = false): [number, number] {
+    const cursor = doc.getTokenCursor(offset);
+    if (cursor.getToken().type === 'open') {
+        return forwardSexpRange(doc);
+    }
+    cursor.forwardList(); // move to the end of the current form
+    const text = doc.model.getText(offset, cursor.offsetStart);
+    const newlineIndex = text.indexOf("\n");
```

Comment:
Do we need to check for `\r` and `\r\n` also?

Reply:
If I'm understanding correctly, `\r\n` would also work with your logic here, since it includes `\n` at the end. You may be able to test this by changing the end of line sequence in a file before running the command for it. You can change the end of line sequence by clicking the button for that in the status bar.

Reply:
Ideally we should test this with some unit tests. Though I don't know if we have the facility to chenge line endings setting for the documents we create there. I think this might be the first non-structural Paredit command we add. As for testing it manually: keep in mind that Calva does not handle changing line ending settings ”in flight”, iirc. You probably need to close and reopen the file. (And we should fix that some day 😄.)

Reply:
I think @bpringe is correct.  My worry about `\r` is somewhat unfounded. Macs switched to `\n` in 2001 with the release of OS X. I'll try testing as you suggest.

Reply:
I saw this PR was moved to draft and then out of draft and a review was requested from us. I just want to check - was this tested on Windows, or in any case, with CLRF/`/r/n` line endings?

Reply:
Yes I tested by clicking on the `LF`/`CRLF` button on the bottom of VS Code, closing and reopening the file.  I will add some tests for this as @PEZ has suggested.  I don't have access to a Windows machine.

Reply:
Added a test for this also.